cylc/uiserver/schema.py

```diff
@@ -248,14 +249,66 @@ class Arguments:
     result = GenericScalar()
 
 
-class UISMutations(Mutations):
+class UISSubscriptions(Subscriptions):
+    # Example subscription:
+    # subscription {
+    #   logs(workflows:["foo"])
+    # }
+
+    async def resolve_logs(
+        root: Optional[Any],
+        info: 'ResolveInfo',
+        *,
+        command='cat_log',
+        workflow: str,
+        task=None,
+        file=None,
+        **kwargs: Any,
+    ) -> AsyncGenerator[Any, None]:
+        """Cat Log Resolver
+        Expands workflow provided subscription query.
+        """
+        parsed_workflows = [Tokens(workflow)]
```

Comment:
We may need to handle the error `Tokens` returns for an invalid ID, not sure. Are errors automatically caught and returned if they happen in this resolver?

Reply:
I just tried this out, the error gets sent through to the UI in a clear manner:

![Screenshot from 2023-01-12 10-16-16](https://user-images.githubusercontent.com/16705946/212040275-440a7f14-930b-4907-b790-56ee8b4c2341.png)

So all good here.